File: app/src/main/java/site/aiok/onepic/model/LevelConfig.kt
package site.aiok.onepic.model

import android.net.Uri

sealed class ImageSource {
    data class Resource(val resId: Int) : ImageSource()
    data class Asset(val path: String) : ImageSource()
    data class UriSource(val uri: Uri) : ImageSource()
    object Generated : ImageSource()
}

data class LevelConfig(
    val levelId: String,
    val title: String,
    val difficulty: String,  // "Easy", "Medium", "Hard"
    val imageSource: ImageSource,
    val rows: Int,           // Number of rows to slice
    val cols: Int,            // Number of columns to slice
    val storyText: String? = null,
    val animationTheme: String? = null,
    val isAscended: Boolean = false,
    // Project Exodus Fields
    val moduleName: String? = null,
    val integrityStatus: String? = null, // e.g. "Quantum Keel: 10% Integrity"
    val motivation: String? = null
)

File: ios/OnePic/OnePic/Model/LevelConfig.swift
import Foundation
import SwiftUI

enum ImageSource {
    case resource(name: String)
    case asset(path: String)
    case url(url: URL)
    case generated
}

struct LevelConfig {
    let levelId: String
    let title: String
    let difficulty: String
    let imageSource: ImageSource
    let rows: Int
    let cols: Int
    var storyText: String? = nil
    var animationTheme: String? = nil
    var isAscended: Bool = false
    // Project Exodus Fields
    var moduleName: String? = nil
    var integrityStatus: String? = nil
    var motivation: String? = nil
}
